vector/src/main/java/im/vector/riotx/features/home/room/detail/composer/TextComposerView.kt

```diff
@@ -158,12 +163,23 @@ class TextComposerView @JvmOverloads constructor(context: Context, attrs: Attrib
         }
     }
 
-    fun setRoomEncrypted(isEncrypted: Boolean) {
-        composerEditText.setHint(
-                if (isEncrypted) {
-                    R.string.room_message_placeholder_encrypted
-                } else {
-                    R.string.room_message_placeholder_not_encrypted
-                })
+    fun setRoomEncrypted(isEncrypted: Boolean, roomEncryptionTrustLevel: RoomEncryptionTrustLevel?) {
+        if (isEncrypted) {
+            composerEditText.setHint(R.string.room_message_placeholder_encrypted)
+            composerShieldImageView.isVisible = true
+            val shield = when (roomEncryptionTrustLevel) {
+                RoomEncryptionTrustLevel.Trusted -> {
+                    ContextCompat.getDrawable(context, R.drawable.ic_shield_trusted)
+                }
+                RoomEncryptionTrustLevel.Warning -> {
+                    ContextCompat.getDrawable(context, R.drawable.ic_shield_warning)
+                }
+                else -> ContextCompat.getDrawable(context, R.drawable.ic_shield_black)
+            }
```

Comment:
A bit of factorisation.
```suggestion
            val shieldResId = when (roomEncryptionTrustLevel) {
                RoomEncryptionTrustLevel.Trusted -> R.drawable.ic_shield_trusted
                RoomEncryptionTrustLevel.Warning -> R.drawable.ic_shield_warning
                else -> R.drawable.ic_shield_black
            }
            val shield = ContextCompat.getDrawable(context, shieldResId)
```

Reply:
updated